pr_agent/algo/pr_processing.py

```diff
@@ -24,10 +24,10 @@ def get_pr_diff(git_provider: Union[GithubProvider, Any], token_handler: TokenHa
     Returns a string with the diff of the PR.
     If needed, apply diff minimization techniques to reduce the number of tokens
     """
-    files = list(git_provider.get_diff_files())
+    git_provider.pr.files = list(git_provider.get_diff_files())
```

Comment:
**Suggestion:** Consider checking if the `pr` object in `git_provider` is not `None` before accessing its `files` attribute. This will prevent potential `AttributeError` if `pr` is `None`.
```suggestion
    if git_provider.pr is not None:
        git_provider.pr.files = list(git_provider.get_diff_files())
```